File: src/main/java/fr/benjamindanlos/laptimes/DTO/SearchDTO.java
package fr.benjamindanlos.laptimes.DTO;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SearchDTO {

	@JsonProperty(value = "track", required = false)
	private String track;

	@JsonProperty(value = "game", required = false)
	private String game;

	@JsonProperty(value = "car", required = false)
	private String car;

	@JsonProperty(value = "driver", required = false)
	private String driver;

	@JsonProperty(value = "searchBest", required = false)
	private Boolean searchBest;

}

File: src/main/kotlin/fr/benjamindanlos/laptimes/DTO/SearchDTO.kt
package fr.benjamindanlos.laptimes.DTO

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Data

@Data
class SearchDTO (

	@JsonProperty(value = "track", required = false)
	val track: String?,

	@JsonProperty(value = "game", required = false)
	val game: String?,

	@JsonProperty(value = "car", required = false)
	val car: String?,

	@JsonProperty(value = "driver", required = false)
	val driver: String?,

	@JsonProperty(value = "searchBest", required = false)
	val searchBest: Boolean?

)
